sharktank/tests/ops/sharded_test.py

```diff
@@ -1366,6 +1366,44 @@ def testGroupNormSplitChannelSharding(self):
         assert expected_bias.is_deep_equal(sharded_theta("bias"), compare_name=False)
 
 
+class Scatter_Test(unittest.TestCase):
+    def setUp(self):
+        torch.random.manual_seed(12345)
+
+    def testScatterReplicatedReplicated(self):
+        tensor = torch.zeros(4, 6, dtype=torch.float32)
+        sharded_tensor = ops.replicate(tensor, count=3)
+        index = torch.randint(0, 4, (2, 1))
+        value = 1
+        index_sharded = ops.replicate(index, count=3)
+        ops.scatter_(sharded_tensor, 1, index_sharded, value)
+        ops.scatter_(tensor, 1, index, value)
+        assert ops.equal(tensor, sharded_tensor)
+
+    def testScatterSplitSplitShardDim(self):
+        tensor = torch.zeros(4, 6, dtype=torch.float32)
+        index = torch.randint(0, 4, (2, 1))
+        value = 1
+        sharded_tensor = ops.reshard_split(tensor, dim=0, count=2)
+        index_sharded = ops.reshard_split(index, dim=0, count=2)
+        ops.scatter_(sharded_tensor, 0, index_sharded, value)
+        ops.scatter_(tensor, 0, index, value)
+        assert ops.equal(tensor, sharded_tensor)
+
+    @parameterized.expand((([3, 1],), ([3, 2],), ([9, 1],), ([9, 6],)))
+    def testScatterSplitSplitNonShardDim(self, index_shape: list[int]):
+        scatter_dim = 1
+        value = 1
+        tensor = torch.zeros(9, 6, dtype=torch.float32)
+        index = torch.randint(0, tensor.shape[scatter_dim], index_shape)
```

Comment:
It seems we can get a conflicting indexing if we have nonunique destination along the scatter axis. Then who wins and gets written last is nondeterministic.
The other 2 tests also do a random index.

You could use [numpy.random.Generator.choice](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.choice.html) with `replace=False` to do a non-repeating random sample.